test/integration/acoustic_comms.cc

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <functional>
+#include <mutex>
+
+#include <gz/msgs.hh>
+#include <gz/transport/Node.hh>
+#include <gz/utils/ExtraTestMacros.hh>
+#include "gz/sim/Server.hh"
+#include "test_config.hh"  // NOLINT(build/include)
+#include "../helpers/EnvTestFixture.hh"
+
+using namespace gz;
+using namespace sim;
+
+/////////////////////////////////////////////////
+class AcousticCommsTest : public InternalFixture<::testing::Test>
+{
+  public:
+    void CommsTestFixture(
+        std::string worldFile,
+        std::string srcAddr, std::string dstAddr,
+        int numMsgs = 3)
```

Comment:
@adityapande-1995 FYI: since you've fully parameterized the test, you could also inherit from `::testing::TestWithParam<AcousticCommsTestDefinition>` where:

```c++
struct AcousticCommsTestDefinition
{
    std::string worldFile;
    std::string srcAddr;
    std::string dstAddr;
    int numMsgs;
};
```

FTR here are the docs for [value-parameterized tests in `gtest`](https://google.github.io/googletest/advanced.html#value-parameterized-tests).

Reply:
Done https://github.com/gazebosim/gz-sim/pull/1608/commits/6aceed49c6a5ef97c9e4484b3eec276b1e2d1a05, looks much better now.